app/models/validation_error.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class ValidationError < ApplicationRecord
+  belongs_to :user
+  validates :form_object, presence: true
+
+  def self.list_of_distinct_errors_with_count
+    distinct_errors = all.flat_map do |e|
+      e.details.flat_map do |attribute, details|
+        details["messages"].map do |message|
+          [e.form_object, attribute, message]
+        end
+      end
+    end
+
+    distinct_errors
+      .tally
+      .sort_by { |_a, b| b }
+      .reverse
```

Comment:
This seems like it might get slow very quickly, with us producing quite a lot of errors across all schools. Do we have any benchmarking from Apply for if we need to look for a quicker alternative or cache the results?

Reply:
I don't know about their benchmark. We can see how it works, and if it gets slow, we can fix it.
Is that ok?